extensions/Aness6040/Aness_FNF_Sprites_Support.js

```diff
@@ -0,0 +1,733 @@
+              xmlText: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: '<xml>...</xml>',
+              },
+            },
+          },    
+          {
+            opcode: 'getXMLAnimationsArray',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'get animations array in [xmlText]',
+            blockIconURI:JsonIcon,
+            arguments: {
+              xmlText: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: '<xml>...</xml>',
+              },
+            },
+          },
+          {
+            opcode: 'getSubTextureCoordinates',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'get [animation] [number] coordinates in [xmlText]',
+            blockIconURI:JsonIcon,
+            arguments: {
+              animation: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: 'Animation Frame',
+              },
+              number: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: '0',
+              },
+              xmlText: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: '<xml>...</xml>',
+              },
+            },
+          },
+          {
+            opcode: 'getSubtexturesArray',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'get subtextures/frames array in [xmlText]',
+            blockIconURI:JsonIcon,
+            arguments: {
+              xmlText: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: '<xml>...</xml>',
+              },
+            },
+          },
+          {
+            opcode: 'haxeflixelextras',
+            blockType: Scratch.BlockType.LABEL,
+            text: 'Haxeflixel Extras',
+          },
+          {
+            opcode: 'convertSize',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'convert haxeflixel size [size] to scratch size',
+            blockIconURI: FlixelIcon,
+            arguments: {
+              size: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: 1,
+              },
+            },
+          }, 
+          {
+            opcode: 'getFlippedImageURI',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'get flipped image uri [uri] x axis: [flipX] y axis: [flipY]',
+            blockIconURI: FlixelIcon,
+            arguments: {
+              uri: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: 'data:image/png;base64, ...',
+              },
+              flipX: {
+                type: Scratch.ArgumentType.BOOLEAN,
+                defaultValue: true,
+              },
+              flipY: {
+                type: Scratch.ArgumentType.BOOLEAN,
+                defaultValue: false,
+              },
+            },
+          },
+          {
+            opcode: 'booleanMenu',
```

Comment:
As opposed to having a separate block for this (that's already duplicated in other extensions), you could let the boolean menu accept reporters, remove (or hide) this block and use the menu as opposed to empty boolean inputs.

Reply:
> As opposed to having a separate block for this (that's already duplicated in other extensions), you could let the boolean menu accept reporters, remove (or hide) this block and use the menu as opposed to empty boolean inputs.

i didnt understand